Javascript/100-Problems/script.js

```diff
@@ -850,3 +850,309 @@ const studentWithLowestPercentageInSubject = (subjectName) => {
 };
 
 console.log(studentWithLowestPercentageInSubject("Maths"));
+
+//function to find and print the subject(s) with the highest/lowest percentage of marks for a specific student
+
+const printSubjectPercentageOfStudent = (
+  studentName,
+  option,
+  obtainedPercentage
+) => {
+  const students = classObj.students;
+  const totalmark = 50;
+  let subjectName = "";
+  let subjectArray = [];
+  let subjectMark = 0;
+
+  students.find((student) => {
+    const subjects = student.marks;
+
+    if (student.name === studentName) {
+      subjectArray = subjects.map((subject) => {
+        subjectName = subject.subject;
+        subjectMark = subject.mark;
+
+        switch (option) {
+          case "highest":
+            if (obtainedPercentage <= subject.mark) {
+              obtainedPercentage = subject.mark;
+            }
+            break;
+
+          case "lowest":
+            if (obtainedPercentage >= subject.mark) {
+              obtainedPercentage = subject.mark;
+            }
+            break;
+
+          default:
+            return "Entered functionality not available! Please Enter the correct one";
+        }
+
+        const percentages = percentage(subjectMark, totalmark);
+        return { studentName, subjectName, percentages };
+      });
```

Comment:
we can also do it like this. No need to declare a variable.
return { studentName, subjectName, percentages: percentage(subjectMark, totalmark) };